testsuite/features/support/env.rb

```diff
@@ -127,31 +127,73 @@ def capybara_register_driver
   $feature_scope = scenario.location.file.split(%r{(\.feature|/)})[-2]
 end
 
-# embed a screenshot after each failed scenario
+# Embed a screenshot after each failed scenario
 After do |scenario|
   current_epoch = Time.new.to_i
   log "This scenario took: #{current_epoch - @scenario_start_time} seconds"
+  log "Current url is #{current_url}"
   if scenario.failed?
     begin
-      Dir.mkdir('screenshots') unless File.directory?('screenshots')
-      path = "screenshots/#{scenario.name.tr(' ./', '_')}.png"
-      # only click on Details when we have errors during bootstrapping and more Details available
-      click_button('Details') if has_content?('Bootstrap Minions') && has_content?('Details')
-      # a Timeout::Error may be raised while a page is still (re)loading
-      find('#page-body', wait: 3) if scenario.exception.is_a?(Timeout::Error)
+      if web_session_is_active?
+        handle_screenshot_and_relog(scenario, current_epoch)
+      else
+        warn 'Page is not visible; unable to take a screenshot.'
+      end
+    ensure
+      print_server_logs
+    end
+  end
+  page.instance_variable_set(:@touched, false)
+end
+
+# Test is web session is open
+def web_session_is_active?
+  # When no Web Session is open, current_url is equal to data:,
+  return false if current_url.empty? || current_url == 'data:,'
+  page.has_selector?('header', wait: 0) || page.has_selector?('#username-field', wait: 0)
+end
+
+# Take a screenshot and try to log back at suse manager server
+def handle_screenshot_and_relog(scenario, current_epoch)
+  Dir.mkdir('screenshots') unless File.directory?('screenshots')
+  path = "screenshots/#{scenario.name.tr(' ./', '_')}.png"
+  Timeout.timeout(Capybara.default_max_wait_time) do
+    begin
+      click_details_if_present
       page.driver.browser.save_screenshot(path)
       attach path, 'image/png'
-      attach "#{Time.at(@scenario_start_time).strftime('%H:%M:%S:%L')} - #{Time.at(current_epoch).strftime('%H:%M:%S:%L')} | Current URL: #{current_url}", 'text/plain'
+      # Attach additional information
+      scenario_start_time_str = Time.at(@scenario_start_time).strftime('%H:%M:%S:%L')
+      current_epoch_str = Time.at(current_epoch).strftime('%H:%M:%S:%L')
+      text_attachment = "#{scenario_start_time_str} - #{current_epoch_str} | Current URL: #{current_url}"
+      attach text_attachment, 'text/plain'
```

Comment:
I changed `attach "#{Time.at(@scenario_start_time).strftime('%H:%M:%S:%L')} - #{Time.at(current_epoch).strftime('%H:%M:%S:%L')} | Current URL: #{current_url}", 'text/plain'` for this block because the former one is crashing the report and screenshot generation.
For me both are doing the same but one will crash and the other not. 
I'm open to feedback from more advance ruby devs.
Result with error : http://10.84.221.160/results/output_20240819051635-build_validation_sanity_check.html

Reply:
I spent a fair amount of time trying to understand this behavior but couldn't explain it.

Reply:
Revert, issue was coming to relog